src/HealthGPS/energy_balance_model.cpp

```diff
@@ -78,131 +82,169 @@ void EnergyBalanceModel::update_risk_factors(RuntimeContext &context) {
             continue;
         }
 
-        simulate_person(person, true, final_shift);
+        // TODO: Compute and update risk factors.
+        simulate_person(person, shift);
+        // SimulatePersonState state = simulate_person(person, shift);
+        // person.risk_factors[H_key] = state.H;
+        // person.risk_factors[BW_key] = state.BW;
+        // person.risk_factors[PAL_key] = state.PAL;
+        // person.risk_factors[RMR_key] = state.RMR;
+        // person.risk_factors[F_key] = state.F;
+        // person.risk_factors[L_key] = state.L;
+        // person.risk_factors[ECF_key] = state.ECF;
+        // person.risk_factors[G_key] = state.G;
+        // person.risk_factors[W_key] = state.W;
+        // person.risk_factors[EE_key] = state.EE;
+        // person.risk_factors[EI_key] = state.EI;
     }
 }
 
-std::array<double, 2> EnergyBalanceModel::simulate_person(Person &person, bool final_run,
-                                                          double final_shift) const {
-    // Model initial state.
+SimulatePersonState EnergyBalanceModel::simulate_person(Person &person, double shift) const {
+    // Initial simulated person state.
     const double H_0 = person.get_risk_factor_value(H_key);
     const double BW_0 = person.get_risk_factor_value(BW_key);
     const double PAL_0 = person.get_risk_factor_value(PAL_key);
     const double F_0 = person.get_risk_factor_value(F_key);
     const double L_0 = person.get_risk_factor_value(L_key);
     const double ECF_0 = person.get_risk_factor_value(ECF_key);
+    const double G_0 = person.get_risk_factor_value(G_key);
     const double EI_0 = person.get_risk_factor_value(EI_key);
     const double CI_0 = person.get_risk_factor_value(CI_key);
```

Comment:
In a similar vein to my comment above, maybe `Person` needs a `get_risk_factor_values()` method which returns a `SimulatePersonState`. Would that make sense?

Reply:
Agree in principle. Thing I wonder is how to make `Person` hold a risk factor struct that's generic enough to handle all models, 1 static and 2 dynamic. I feel like we might fall back into the trap of having to save different state objects, with accompanying getter/setter/support code, for each different model, like in that `CachedRepository` issue. Would like to keep risk factor models as decoupled as possible, ideally.

Maybe the getter/setter/support code can all go in `EnergyBalanceModel`, though? How do you see it?

Reply:
Ah yes, of course. I wasn't thinking straight.

You could add getters and setters to `EnergyBalanceModel` if you like. Could just be a freestanding function in `energy_balance_model.cpp` too. I just thought it would be cleaner to encapsulate this bit of boilerplate somehow.